pkg/client/client.go

```diff
@@ -151,27 +154,51 @@ func (w *worker) run(ctx context.Context) {
 // Client is a struct which is used the run workload to a specific DB.
 type Client struct {
 	p        *properties.Properties
-	wg       sync.WaitGroup
 	workload ycsb.Workload
-	db       ycsb.DB
+	db       DbWrapper
 }
 
 // NewClient returns a client with the given workload and DB.
 // The workload and db can't be nil.
-func NewClient(p *properties.Properties, workload ycsb.Workload, db ycsb.DB) *Client {
+func NewClient(p *properties.Properties, workload ycsb.Workload, db DbWrapper) *Client {
 	return &Client{p: p, workload: workload, db: db}
 }
 
 // Run runs the workload to the target DB, and blocks until all workers end.
 func (c *Client) Run(ctx context.Context) {
+	var wg sync.WaitGroup
 	threadCount := c.p.GetInt(prop.ThreadCount, 100)
 
-	c.wg.Add(threadCount)
+	wg.Add(threadCount)
+	measureCtx, measureCancel := context.WithCancel(ctx)
+	go func() {
+		// load stage no need to warm up
+		if c.p.GetBool(prop.DoTransactions, true) {
+			dur := c.p.GetInt64("warmuptime", 0)
+			time.Sleep(time.Duration(dur) * time.Second)
+		}
+		// finish warming up
+		c.db.WarmUp = false
```

Comment:
data race for WarmUp var

Reply:
I have noticed that, but it doesn't matter a lot. `WarmUp` is only set once here and for read threads it is acceptable to get a slightly old value.

Reply:
use atomic to avoid data race.